code/tasks/UserFormsUpgradeService.php

```diff
@@ -224,4 +230,66 @@ public function getQuiet()
     {
         return $this->quiet;
     }
+
+
+    /**
+     * Migrate Placeholder data from field specific table to the EditableFormField table
+     *
+     * @param EditableFormField $field
+     * @param string $tableName
+     */
+    private function migratePlaceholder($field, $tableName)
+    {
+        // Migrate Placeholder setting from $tableName table to EditableFormField table
+        if ($field->Placeholder) {
+            return;
+        }
+        // Check if draft table exists
+        $query = "SHOW TABLES LIKE '$tableName'";
+        $tableExists = DB::query($query)->value();
+        if ($tableExists == null) {
+            // Check if _obsolete_ draft table exists
+            $tableName = '_obsolete_' . $tableName;
+            $query = "SHOW TABLES LIKE '$tableName'";
+            $tableExists = DB::query($query)->value();
+            if ($tableExists == null) {
+                return;
+            }
+        }
+        // Check if old Placeholder column exists
+        $query = "SHOW COLUMNS FROM `$tableName` LIKE 'Placeholder'";
+        $columnExists = DB::query($query)->value();
+        if ($columnExists == null) {
+            return;
+        }
+        // Fetch existing draft Placeholder value
+        $query = "SELECT `Placeholder` FROM `$tableName` WHERE `ID` = '$field->ID'";
+        $draftPlaceholder = DB::query($query)->value();
+
+        if (!$draftPlaceholder) {
+            return;
+        }
+        // Update draft Placeholder value
+        $query = "UPDATE `EditableFormField` SET `Placeholder` = '$draftPlaceholder' WHERE `ID` = '$field->ID'";
```

Comment:
Can you please SQL escape this? there's a SQL injection vulnerability here.

If possible, use DB::prepared_query, then you can use placeholders instead.

```php
DB::prepared_query(
	"UPDATE \"EditableFormField\" SET \"Placeholder\" = ? WHERE "ID" = ?",
	[ $draftPlaceholder, $field->ID]
);
```

Please use double quotes instead of back ticks if you can help it. ticks are mysql-only.